### ISSUE TITLE
Errors from deprecated DUMPI_CONST while building sst-dumpi

### ISSUE THREAD

**Issue by lenakanellou** [sstsimulator/sst-dumpi#28]
Hi, 

I am trying to install sst-dumpi on a Ubuntu 18.04.5 LTS system with a 5.7.9 kernel version. 
I have downloaded the current master version (tagged with 7.1.0) and I have followed the configuration instructions, meaning that I've specified the --enable-libdumpi option and have specified CC and CXX parameters for make. Specifically, for CC / CXX I have tried mpicc and mpicxx from the 4.0, 1.8.1, and 1.10.7 OpenMPI codebases. 

The configuration finishes fine, but I cannot build the system: the make fails in all versions of OpenMPI.

I get "conflicting types" errors for functions included both from the dumpi code as well as the OpenMPI code, and I've noticed that a type specification of dumpi is involved. Specifically: DUMPI_DEPRECATED_CONST. 

I am attaching error logs with more information. 

My issue is simlar to issues 5 and 7, that are already closed: 
https://github.com/sstsimulator/sst-dumpi/issues/5
https://github.com/sstsimulator/sst-dumpi/issues/7

However, even when I apply the solutions proposed in there, I keep having the same problems. 

Any hints on how I should proceed to have a successful installation?

Thanks!

[error-4.0.log](https://github.com/sstsimulator/sst-dumpi/files/6802299/error-4.0.log)
[error-1.10.7.log](https://github.com/sstsimulator/sst-dumpi/files/6802301/error-1.10.7.log)
[error-1.8.1.log](https://github.com/sstsimulator/sst-dumpi/files/6802302/error-1.8.1.log)

**Comment by jpkenny** [comment]
Sorry for the slow reply. DUMPI isn't being actively developed and we're stretched pretty thin at the moment for supporting the package.

I can reproduce your error with openmpi-4.1.0 and it looks like there are now asserts enforcing MPI 3 compliance that are breaking dumpi. It'll take me some time to implement a fix for this. In the meantime, you should be able to use this patch to get things running (yes, it's an ugly hack...):

```
diff --git a/acinclude/mpitypes.m4 b/acinclude/mpitypes.m4
index 9fe2553..142818e 100644
--- a/acinclude/mpitypes.m4
+++ b/acinclude/mpitypes.m4
@@ -77,12 +77,8 @@ AC_DEFINE(DUMPI_HAVE_MPI_PACKED, [1])
 AC_DEFINE(DUMPI_DATATYPE_MPI_PACKED, [MPI_PACKED])
 AH_TEMPLATE([DUMPI_HAVE_MPI_LB], [defined if MPI_LB is available])
 AH_TEMPLATE([DUMPI_DATATYPE_MPI_LB], [Map DUMPI types to MPI])
-AC_DEFINE(DUMPI_HAVE_MPI_LB, [1])
-AC_DEFINE(DUMPI_DATATYPE_MPI_LB, [MPI_LB])
 AH_TEMPLATE([DUMPI_HAVE_MPI_UB], [defined if MPI_UB is available])
 AH_TEMPLATE([DUMPI_DATATYPE_MPI_UB], [Map DUMPI types to MPI])
-AC_DEFINE(DUMPI_HAVE_MPI_UB, [1])
-AC_DEFINE(DUMPI_DATATYPE_MPI_UB, [MPI_UB])
 AH_TEMPLATE([DUMPI_HAVE_MPI_FLOAT_INT], [defined if MPI_FLOAT_INT is available])
 AH_TEMPLATE([DUMPI_DATATYPE_MPI_FLOAT_INT], [Map DUMPI types to MPI])
 AC_DEFINE(DUMPI_HAVE_MPI_FLOAT_INT, [1])
@@ -107,4 +103,4 @@ AH_TEMPLATE([DUMPI_HAVE_MPI_LONG_DOUBLE_INT], [defined if MPI_LONG_DOUBLE_INT is
 AH_TEMPLATE([DUMPI_DATATYPE_MPI_LONG_DOUBLE_INT], [Map DUMPI types to MPI])
 AC_DEFINE(DUMPI_HAVE_MPI_LONG_DOUBLE_INT, [1])
 AC_DEFINE(DUMPI_DATATYPE_MPI_LONG_DOUBLE_INT, [MPI_LONG_DOUBLE_INT])
-])
\ No newline at end of file
+])
diff --git a/configure.ac b/configure.ac
index 1220653..9fdff20 100644
--- a/configure.ac
+++ b/configure.ac
@@ -137,6 +137,17 @@ else
 fi
 AC_SUBST([HAVE_DOT])
 
+AC_DEFINE(DUMPI_SKIP_MPI_TYPE_HVECTOR, [1])
+AC_DEFINE(DUMPI_SKIP_MPI_TYPE_HINDEXED, [1])
+AC_DEFINE(DUMPI_SKIP_MPI_TYPE_STRUCT, [1])
+AC_DEFINE(DUMPI_SKIP_MPI_ADDRESS, [1])
+AC_DEFINE(DUMPI_SKIP_MPI_ERRHANDLER_CREATE, [1])
+AC_DEFINE(DUMPI_SKIP_MPI_ERRHANDLER_SET, [1])
+AC_DEFINE(DUMPI_SKIP_MPI_ERRHANDLER_GET, [1])
+AC_DEFINE(DUMPI_SKIP_MPI_TYPE_EXTENT, [1])
+AC_DEFINE(DUMPI_SKIP_MPI_TYPE_LB, [1])
+AC_DEFINE(DUMPI_SKIP_MPI_TYPE_UB, [1])
+
 AC_CONFIG_FILES([
  Makefile
  dumpi/Makefile
diff --git a/dumpi/libdumpi/data.c b/dumpi/libdumpi/data.c
index 7d2de76..f21484d 100644
--- a/dumpi/libdumpi/data.c
+++ b/dumpi/libdumpi/data.c
@@ -70,8 +70,6 @@ const MPI_Datatype dumpi_mpi_datatype_match[DUMPI_FIRST_USER_DATATYPE] = {
   DUMPI_DATATYPE_MPI_UNSIGNED_LONG_LONG,
   DUMPI_DATATYPE_MPI_LONG_LONG,
   DUMPI_DATATYPE_MPI_PACKED,
-  DUMPI_DATATYPE_MPI_LB,
-  DUMPI_DATATYPE_MPI_UB,
   DUMPI_DATATYPE_MPI_FLOAT_INT,
   DUMPI_DATATYPE_MPI_DOUBLE_INT,
   DUMPI_DATATYPE_MPI_LONG_INT,
@@ -96,4 +94,4 @@ const MPI_Op dumpi_mpi_op_match[DUMPI_FIRST_USER_OP] = {
   DUMPI_OP_MPI_MAXLOC,
   DUMPI_OP_MPI_REPLACE,
   MPI_OP_NULL
-};
\ No newline at end of file
+};
diff --git a/dumpi/libdumpi/mpibindings.h b/dumpi/libdumpi/mpibindings.h
index 1757bd1..6367e22 100644
--- a/dumpi/libdumpi/mpibindings.h
+++ b/dumpi/libdumpi/mpibindings.h
@@ -345,27 +345,18 @@ extern "C" {
 #if DUMPI_HAVE_MPI_COMBINER_VECTOR
     case MPI_COMBINER_VECTOR: return DUMPI_COMBINER_VECTOR;
 #endif
-#if DUMPI_HAVE_MPI_COMBINER_HVECTOR_INTEGER
-    case MPI_COMBINER_HVECTOR_INTEGER: return DUMPI_COMBINER_HVECTOR_INTEGER;
-#endif
 #if DUMPI_HAVE_MPI_COMBINER_HVECTOR
     case MPI_COMBINER_HVECTOR: return DUMPI_COMBINER_HVECTOR;
 #endif
 #if DUMPI_HAVE_MPI_COMBINER_INDEXED
     case MPI_COMBINER_INDEXED: return DUMPI_COMBINER_INDEXED;
 #endif
-#if DUMPI_HAVE_MPI_COMBINER_HINDEXED_INTEGER
-    case MPI_COMBINER_HINDEXED_INTEGER: return DUMPI_COMBINER_HINDEXED_INTEGER;
-#endif
 #if DUMPI_HAVE_MPI_COMBINER_HINDEXED
     case MPI_COMBINER_HINDEXED: return DUMPI_COMBINER_HINDEXED;
 #endif
 #if DUMPI_HAVE_MPI_COMBINER_INDEXED_BLOCK
     case MPI_COMBINER_INDEXED_BLOCK: return DUMPI_COMBINER_INDEXED_BLOCK;
 #endif
-#if DUMPI_HAVE_MPI_COMBINER_STRUCT_INTEGER
-    case MPI_COMBINER_STRUCT_INTEGER: return DUMPI_COMBINER_STRUCT_INTEGER;
-#endif
 #if DUMPI_HAVE_MPI_COMBINER_STRUCT
     case MPI_COMBINER_STRUCT: return DUMPI_COMBINER_STRUCT;
 #endif
@@ -955,4 +946,4 @@ extern "C" {
 } /* End of extern C block */
 #endif /* ! __cplusplus */
 
-#endif /* DUMPI_LIBDUMPI_MPIBINDINGS_H */
\ No newline at end of file
+#endif /* DUMPI_LIBDUMPI_MPIBINDINGS_H */
```

**Comment by jpkenny** [comment]
fixed in devel via PR #29
